projects/suricata/build.sh

```diff
@@ -0,0 +1,271 @@
+
+echo "[x] Libfuzzer stage"
+export CC=clang
+export CXX=clang++
+export ac_cv_func_malloc_0_nonnull=yes
+export ac_cv_func_realloc_0_nonnull=yes
+export CFLAGS="-g -fsanitize=fuzzer-no-link,address,integer,bounds,null,undefined,float-divide-by-zero"
+export CXXFLAGS="-g -fsanitize=fuzzer-no-link,address,integer,bounds,null,undefined,float-divide-by-zero"
+export LDFLAGS="-fsanitize=address,integer,bounds,null,undefined,float-divide-by-zero"
+export RUSTFLAGS="-Cpasses=sancov-module -Cllvm-args=-sanitizer-coverage-level=4 -Cllvm-args=-sanitizer-coverage-trace-compares"
+export RUSTFLAGS="$RUSTFLAGS -Cllvm-args=-sanitizer-coverage-inline-8bit-counters -Cllvm-args=-sanitizer-coverage-pc-table -Clink-dead-code -Cllvm-args=-sanitizer-coverage-stack-depth -Ccodegen-units=1 -Cdebug-assertions=yes"
+export CARGO_BUILD_TARGET="x86_64-unknown-linux-gnu"
+
+(
+cd pcre2-10.39
+./configure --disable-shared
+make -j$(nproc) clean
+make -j$(nproc) all
+make -j$(nproc) install
+)
+
+(
+cd lz4-1.9.2
+make liblz4.a
+cp lib/liblz4.a /usr/local/lib/
+cp lib/lz4*.h /usr/local/include/
+)
+
+(
+cd jansson-2.12
+./configure --disable-shared
+make -j$(nproc)
+make install
+)
+
+(
+cd fuzzpcap
+mkdir build
+cd build
+cmake ..
+make install
+)
+
+(
+cp -r libhtp suricata && cd suricata
+
+sh autogen.sh
+./src/tests/fuzz/oss-fuzz-configure.sh && make -j$(nproc)
+
+mkdir libfuzzer
+
+./src/suricata --list-app-layer-protos | tail -n +2 | while read i; do cp src/fuzz_applayerparserparse libfuzzer/fuzz_applayerparserparse_$i; done
+
+mv libfuzzer/fuzz_applayerparserparse_http libfuzzer/fuzz_applayerparserparse_http1
+
+cp src/fuzz_applayerprotodetectgetproto libfuzzer/fuzz_applayerprotodetectgetproto
+cp src/fuzz_mimedecparseline libfuzzer/fuzz_mimedecparseline
+cp src/fuzz_predefpcap_aware libfuzzer/fuzz_predefpcap_aware
+cp src/fuzz_sigpcap_aware libfuzzer/fuzz_sigpcap_aware
+
+make clean
+)
+
+echo "[x] AFL++ stage"
+export CC=afl-clang-lto
+export CXX=afl-clang-lto++
+export ac_cv_func_malloc_0_nonnull=yes
+export ac_cv_func_realloc_0_nonnull=yes
+export CFLAGS="-g -fsanitize=address,integer,bounds,null,undefined,float-divide-by-zero"
+export CXXFLAGS="-g -fsanitize=address,integer,bounds,null,undefined,float-divide-by-zero"
+export RUSTFLAGS="-C debug-assertions -C overflow_checks -C passes=sancov-module -C codegen-units=1 -C llvm-args=-sanitizer-coverage-level=3 -C llvm-args=-sanitizer-coverage-trace-pc-guard -C llvm-args=-sanitizer-coverage-prune-blocks=0 -C opt-level=3 -C target-cpu=native -Clink-arg=-fuse-ld=gold"
+
+(
+cd pcre2-10.39
+./configure --disable-shared
+make -j$(nproc) clean
+make -j$(nproc) all
+make -j$(nproc) install
+)
+
+(
+cd lz4-1.9.2
+make liblz4.a
+cp lib/liblz4.a /usr/local/lib/
+cp lib/lz4*.h /usr/local/include/
+)
+
+(
+cd jansson-2.12
+make clean
+./configure --disable-shared
+make -j$(nproc)
+make install
+)
+
+(
+cd fuzzpcap
+rm -rf build
+mkdir build
+cd build
+cmake ..
+make install
+)
+
+(
+cp -r libhtp suricata && cd suricata
```

Comment:
`rm -rf suricata/libhtp` before copying. (But anyway libhtp is cleaned during suricata's `make clean`)